### ISSUE TITLE
allow setting star ratings from overview

### ISSUE THREAD

**Issue by foss-** [vmiklos/plees-tracker#454]
**Reproducer steps**
1. when on overview page and looking to set a star rating for a certain night in the past sleeps listing, when tapping the stars...

**Actual result**
... a separate page opens.

**Expected behavior**
... opening a separate page really isn't needed and complicates the interaction unnecessarily. It would be easier if setting the star rating was possible from the Past sleeps overview page.

**Comment by vmiklos** [comment]
Yes, this is indeed a problem. The trouble is that it seems the intention is that no interactive items are included in an item of a recycler view, which goes against the idea of interacting with the rating there.

See #161, do you have an idea how to do what you want in a way that doesn't re-introduce that issue?

**Comment by foss-** [comment]
Sorry I have no understanding of how recycler view works. If this is a limitation, setting stars on that view probably won't be feasible.

From a UX standpoint, preventing a reintroduction of #161 seems feasible via two mechanisms:

- make setting the star rating require a long-press (unsure if scrolling, the finger remains on the UI element but it moves counts as long press - intuitively I'd say no, but needs to be verified. Only after long-pressing initiate the UI to set / change star rating by switching the long-press into swipe gesture)
or
- when tapping the stars show an overlay that requires additional interaction to set the star rating (this could work in a way, that if no further interaction is done, e.g. when intending to scroll, the overlay automatically closes once the user lifts their finger. Setting a star rating could require an additional lift of the finger and additional tap on the star rating in the overlay)

Wondering how other Android apps handle star ratings. I had a glance at my installed apps but none used star ratings. Probalby the commercial sleep trackers do. But then again those are not open source so can't be inspected easily as to how their implementation works.

**Comment by RutvikLathiya27** [comment]
is This bug still open?

**Comment by foss-** [comment]
yes, as tapping stars does not set the rating, but open the detailed view. Obviously there the star rating can be set.

Since this generally works this is more of a UX / qualitiy of life improvement and low priority.

**Comment by vmiklos** [comment]
Right, see above :) There was an earlier explicit request to not modify the rating in the main activity, as it interacts with vertical scrolling of the sleep items. So setting the rating from there is currently disabled, till we find a way to do this without introducing a regression.

Creative ideas are welcome. I'm not sure yet another setting is the way to go here.

**Comment by RutvikLathiya27** [comment]
Ok i will try to fix that.

**Comment by vmiklos** [comment]
I guess the primary use-case is that you swipe up/down to navigate the list of sleep items or you swipe left/right to delete a sleep item. So perhaps a trick you could implement is to re-introduce hit testing, but only set the rating on tap, to not break the swipe in the different directions.

See 6198c6fae114425fd455259666edff9644a6d1d0 that removed the hit test code that was in place previously (that even routed the swipe to the rating bar, so it's not wanted as-is).
